docs/upgrade.html

```diff
@@ -71,17 +71,31 @@ <h4><a id="upgrade_2_4_0" href="#upgrade_2_4_0">Upgrading from 0.8.x, 0.9.x, 0.1
 <p><b>Additional Upgrade Notes:</b></p>
 
 <ol>
-    <li>ZooKeeper has been upgraded to 3.5.6. Set <code>snapshot.trust.empty=true</code> in <code>zookeeper.properties</code> before the upgrade
+    <li>ZooKeeper has been upgraded to 3.5.6. ZooKeeper upgrade from 3.4.X to 3.5.6 can fail if there are no snapshot files in 3.4 data directory.
+        This usually happens in test upgrades where ZooKeeper 3.5.6 is trying to load an existing 3.4 data dir in which no snapshot file has been created.
+        For more details about the issue please refer to <a href="https://issues.apache.org/jira/browse/ZOOKEEPER-3056">ZOOKEEPER-3056</a>.
+        For multi-node ZooKeeper cluster upgrades, set <code>snapshot.trust.empty=true</code> in <code>zookeeper.properties</code> before the upgrade
```

Comment:
I'm not sure why we would ever recommend setting `snapshot.trust.empty`, when we know that it is buggy.  We should explicitly tell them not to do this, then link to the ZK JIRA explaining why not.  Then we should explain what the right solution is (downloading the blank snapshot file)

Reply:
We have observed data loss only in standalone cluster upgrades. So I have recommended  `snapshot.trust.empty` flag for multi-node cluster clusters.  I have updated the note to avoid unnecessary confusion.

@cmccabe @ijuma  Can you take look please.. I would like to include the upgrade for 2.4.0 release.